File: fixtures/scale/lua/large/checks/check_riverstream.lua
local riverstream = require("ciphers.riverstream")
local riverstream_spec = require("specs.riverstream_spec")
local corpus_stream = require("corpus.corpus_stream")

local function check_riverstream()
  local spec = riverstream_spec.get()
  for _, text in ipairs(corpus_stream.corpus_stream()) do
    local sealed = riverstream.riverstream_encrypt(text, spec.key)
    local opened = riverstream.riverstream_decrypt(sealed, spec.key)
    if opened ~= text then return false end
  end
  return spec.category == "stream"
end

return { check_riverstream = check_riverstream }

File: fixtures/scale/typescript/large/src/checks/CheckRiverstream.ts
import { Riverstream } from '../ciphers/Riverstream';
import { RiverstreamSpec } from '../specs/RiverstreamSpec';
import { CorpusStream } from '../corpus/CorpusStream';

export class CheckRiverstream {
  static run(): boolean {
    const spec = RiverstreamSpec.get();
    for (const text of CorpusStream.texts()) {
      const sealed = Riverstream.encrypt(text, spec.key);
      const opened = Riverstream.decrypt(sealed, spec.key);
      if (opened !== text) return false;
    }
    return spec.category === 'stream';
  }
}
